src/backend/distributed/utils/background_jobs.c

```diff
@@ -706,8 +706,12 @@ TaskConcurrentCancelCheck(TaskExecutionContext *taskExecutionContext)
 	BackgroundExecutorHashEntry *handleEntry = taskExecutionContext->handleEntry;
 	BackgroundTask *task = GetBackgroundTaskByTaskId(handleEntry->taskid);
 	taskExecutionContext->task = task;
+	if (!task)
+	{
+		ereport(ERROR, (errmsg("task is invalid")));
+	}
```

Comment:
You message is not very helpful.

See around line 300 a similar pattern, here patched:

```
		if (!task)
		{
			ereport(ERROR, (errmsg("no task found with taskid: %ld", handleEntry->taskid)));
		}
```

Also I suggest to move your patch one line up (before the assignment).

Reply:
I wonder if the error message should not be more strong: 'unexpected missing task id: ...;'

Maybe an Assert() given "task" is always expected to be set....

Reply:
I'm unsure what Citus guidelines are here but I don't think task is ever supposed to be NULL in this function.
That's why maybe an Assert() and not an ereport(ERROR)...

Reply:
I'm not sure whether the use of the "assert" is more appropriate, but the ereport(ERROR) is a normal shutdown in the event of errors.